### ISSUE TITLE
Should Mypy warn about potential invalid arguments to `max`?

### ISSUE THREAD

**Issue by willmcgugan** [python/mypy#8863]
I was surprised that Mypy didn't flag an issue where I was calling `max` with an int and a None, which throws a TypeError at runtime.

For example, in the following code, should I not expect Mypy to warn that `top` is potentially None, which wouldn't work with `max` ?

```python
from typing import Optional
top: Optional[int] = None
print(max(5, top))
```

**Comment by JelleZijlstra** [comment]
This is perhaps more of a typeshed issue. Currently `max()` is stubbed as accepting basically anything (https://github.com/python/typeshed/blob/master/stdlib/2and3/builtins.pyi#L1318). Perhaps it could be changed to accept only types that define `<`.
